dialogue-clients/src/main/java/com/palantir/dialogue/clients/DefaultDialogueDnsResolver.java

```diff
@@ -41,8 +49,81 @@ public ImmutableSet<InetAddress> resolve(String hostname) {
             }
             return ImmutableSet.copyOf(results);
         } catch (UnknownHostException e) {
-            log.warn("Unknown host '{}'", UnsafeArg.of("hostname", hostname), e);
+            GaiError gaiError = extractGaiErrorString(e);
+            log.warn(
+                    "Unknown host '{}'",
+                    SafeArg.of("gaiErrorType", gaiError.name()),
+                    SafeArg.of("gaiErrorMessage", gaiError.errorMessage()),
+                    UnsafeArg.of("hostname", hostname),
+                    e);
+            metrics.lookupError(gaiError.name()).mark();
             return ImmutableSet.of();
         }
     }
+
+    // these strings were taken from glibc-2.39, but likely have not changed in quite a while
+    // strings may be different on BSD systems like macos
+    // TODO(dns): update this list to try to match against known strings on other platforms
+    private enum GaiError {
+        EAI_ADDRFAMILY("Address family for hostname not supported"),
+        EAI_AGAIN("Temporary failure in name resolution"),
+        EAI_BADFLAGS("Bad value for ai_flags"),
+        EAI_FAIL("Non-recoverable failure in name resolution"),
+        EAI_FAMILY("ai_family not supported"),
+        EAI_MEMORY("Memory allocation failure"),
+        EAI_NODATA("No address associated with hostname"),
+        EAI_NONAME("Name or service not known"),
+        EAI_SERVICE("Servname not supported for ai_socktype"),
+        EAI_SOCKTYPE("ai_socktype not supported"),
+        EAI_SYSTEM("System error"),
+        EAI_INPROGRESS("Processing request in progress"),
+        EAI_CANCELED("Request canceled"),
+        EAI_NOTCANCELED("Request not canceled"),
+        EAI_ALLDONE("All requests done"),
+        EAI_INTR("Interrupted by a signal"),
+        EAI_IDN_ENCODE("Parameter string not correctly encoded"),
+        EAI_OVERFLOW("Result too large for supplied buffer"),
+        CACHED(),
+        UNKNOWN();
+
+        private final Optional<String> errorMessage;
+
+        GaiError() {
+            this.errorMessage = Optional.empty();
+        }
+
+        GaiError(String errorMessage) {
+            this.errorMessage = Optional.of(errorMessage);
+        }
+
+        @Safe
+        String errorMessage() {
+            return errorMessage.orElseGet(this::name);
+        }
+    }
+
+    private static GaiError extractGaiErrorString(UnknownHostException exception) {
```

Comment:
perhaps `extractGaiError` since this returns `GaiError`, not `String`

Reply:
done